### ISSUE TITLE
OCommandSQLParsingException when edge type is named Edge

### ISSUE THREAD

**Issue by obi1kenobi** [orientechnologies/orientdb#5814]
```
CREATE CLASS Test EXTENDS V
CREATE PROPERTY Test.name String

# I'd accept an error message here, saying that Edge is a reserved keyword
# and cannot be the name of a class, but no such error happens
CREATE CLASS Edge EXTENDS E
```

Then, the following query causes the exception below:
```
CREATE EDGE Edge FROM (SELECT FROM Test WHERE name='a') TO (SELECT FROM Test WHERE name='b')
```

```
com.orientechnologies.orient.core.sql.OCommandSQLParsingException: Error on parsing command at position #0: Error parsing query: CREATE EDGE Edge FROM (SELECT FROM Test WHERE name='a') TO (SELECT FROM Test WHERE name='b') Encountered " <EDGE> "Edge "" at line 1, column 13. Was expecting one of: <FROM> ... <TO> ... <VALUES> ... <SET> ... <ADD> ... <PUT> ... <MERGE> ... <CONTENT> ... <REMOVE> ... <ORDER> ... <GROUP> ... <OFFSET> ... <RECORD> ... <LUCENE> ... <NEAR> ... <WITHIN> ... <EXCEPTION> ... <IN> ... <KEY> ... <IDENTIFIER> ... <QUOTED_IDENTIFIER> ...
```

**Comment by luigidellaquila** [comment]
Hi @obi1kenobi 

This is because of the new strict parser, now "edge" is a reserved keyword, you have to back-tick quote it:

```
CREATE EDGE `Edge` FROM (SELECT FROM Test WHERE name='a') TO (SELECT FROM Test WHERE name='b')
```

(or just disable strictSql)

In v.2.2 also DDL statements are strictly parsed, so also the CREATE CLASS statement has to be quoted:
```
CREATE CLASS `Edge` EXTENDS E
```
Thanks

Luigi

**Comment by obi1kenobi** [comment]
Good to know this is fully fixed in 2.2, it's confusing that it currently fails in one setting (create edge) and succeeds in another (create class).
